File: Agents/Interaction/Templates/GetInputFromUser.cs
namespace Technologai.Templates
{
    public class GetInputFromUser : Template
    {
        public GetInputFromUser()
        {
            Id = "get_input_from_user";
            Description = "Receive a text input from the user.";
        }

        public override Task<bool> Assess(Information information) => Task.FromResult(true);

        public override async Task<Data?> Process(Information information)
        {
            return await Task.Run(() =>
            {
                return new Data(Console.ReadLine() ?? string.Empty);
            });
        }
    }
}

File: Agents/Interaction/python_templates/get_input_from_user.py
from technologai_agent.information import Information
from technologai_agent.template import Template

class GetInputFromUser(Template):
    def __init__(self, **kwargs):
        super().__init__(
            id="get_input_from_user",
            description="Receive a text input from the user.",
            **kwargs
        )

    async def assess(self, information: Information):
        return True

    async def process(self, information: Information):
        return {"message": input()}
